### PR TITLE
create root note when site is created

### PR DESCRIPTION
**Pull request by alavieille**
[OO-DEPRECATED] ``initializeNewNode``  is deprecated and replace by ``initializeNode`` in class ``OpenOrchestra\Backoffice\Manager\NodeManager``. this method will be removed in 1.2.0
[OO-FEATURE] When a website is created, an homepage for this site is also created.

https://github.com/open-orchestra/open-orchestra-cms-bundle/pull/1518
https://github.com/open-orchestra/open-orchestra-model-interface/pull/169

### CODE REVIEW

File: Backoffice/EventSubscriber/CreateRootNodeSubscriber.php [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+     * @param ObjectManager               $objectManager
+     * @param TranslatorInterface         $translator
+     */
+    public function __construct(
+        NodeManager $nodeManager,
+        TemplateRepositoryInterface $repositoryTemplate,
+        ObjectManager $objectManager,
+        TranslatorInterface $translator
+    )
+    {
```
Comment (itkg-ngilain) [inline Backoffice/EventSubscriber/CreateRootNodeSubscriber.php]: ``){``

File: Backoffice/EventSubscriber/CreateRootNodeSubscriber.php [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+     */
+    public function createRootNode(SiteEvent $siteEvent)
+    {
+        $templateId = $siteEvent->getTemplateIdNodeHome();
+        $site = $siteEvent->getSite();
+        if (null !== $templateId && null !== $site) {
+            $language = $site->getDefaultLanguage();
+            $template = $this->repositoryTemplate->findOneByTemplateId($templateId);
+            $name = $this->translator->trans('open_orchestra_backoffice.node.root_name');
+            $routePattern = '/';
```
Comment (itkg-ngilain) [inline Backoffice/EventSubscriber/CreateRootNodeSubscriber.php]: Can you put this in a constant to get it factorized and easily updatable ?

File: BackofficeBundle/Resources/translations/messages.en.yml [hunk 3]
```diff
@@ -117,6 +117,9 @@ open_orchestra_backoffice:
             meta_index: Meta index
             meta_follow: Meta follow
             robots_txt: Informations contained in the robots.txt file
+            template_node_root:
+                label: Homepage template
+                helper: Template used to create homepage of your site.
```
Comment (itkg-ngilain) [inline BackofficeBundle/Resources/translations/messages.en.yml]: to create the homepage of your site

File: BackofficeBundle/Resources/translations/messages.fr.yml [hunk 4]
```diff
@@ -117,6 +117,9 @@ open_orchestra_backoffice:
             index: Meta index
             follow: Meta follow
             robots_txt: Informations contenues dans le fichier robots.txt
+            template_node_root:
+                label: Temlate de la page d'accueil
```
Comment (itkg-ngilain) [inline BackofficeBundle/Resources/translations/messages.fr.yml]: Gabarit de la page d'accueil

File: BackofficeBundle/Resources/translations/messages.fr.yml [hunk 5]
```diff
@@ -117,6 +117,9 @@ open_orchestra_backoffice:
...
+                helper: Template utilisé pour la génération de la page d'accueil de votre site.
```
Comment (itkg-ngilain) [inline BackofficeBundle/Resources/translations/messages.fr.yml]: Gabarit utilisé

**Comment by alavieille** [comment]
@itkg-ngilain pr updated

**Comment by itkg-ngilain** [comment]
:+1:

**Pull request merged**